File: src/main/java/de/fluxparticle/animation/signal/SignalJump.java
package de.fluxparticle.animation.signal;

public class SignalJump<T> extends SignalPath<T> {

    public static class Builder<T> extends SignalPath.Builder<T> {

        private final double jumpTime;

        public Builder() {
            this(0.5);
        }

        public Builder(double jumpTime) {
            this.jumpTime = jumpTime;
        }

        @Override
        public SignalPath<T> build() {
            return new SignalJump<>(start, end, jumpTime);
        }

    }

    private final double jumpTime;

    private SignalJump(T start, T end, double jumpTime) {
		super(start, end);
        this.jumpTime = jumpTime;
    }

	@Override
	public T apply(Double t) {
		return t < jumpTime ? start : end;
	}

}

File: src/main/kotlin/de/fluxparticle/animation/signal/SignalJump.kt
package de.fluxparticle.animation.signal

class SignalJump<T : Any> private constructor(start: T, end: T, private val jumpTime: Double) : SignalPath<T>(start, end) {

    class Builder<T : Any> @JvmOverloads constructor(private val jumpTime: Double = 0.5) : SignalPath.Builder<T>() {

        override fun build(): SignalPath<T> {
            return SignalJump(start, end, jumpTime)
        }

    }

    override fun apply(t: Double): T {
        return if (t < jumpTime) start else end
    }

}
